### DISCUSSION TITLE
How to implement sliding window training？

### DISCUSSION THREAD

**Discussion by lanslotttTT** [Project-MONAI/MONAI#4268] [Q&A] [answered]
Thanks for releasing such a great job! I found that sliding window inference has been implemented ,so how to implement sliding window training?

**Comment by rijobro** [comment, marked as answer]
Generally speaking you don't want to use sliding window inference for training, it's non-differentiable. Typically we use random crops for training and then use sliding window inference for validation/testing/deployment.

**Comment by dongyang0122** [comment]
@lanslotttTT can you share more details about sliding-window training? is it similar to the sliding-window scheme in swin transformer?

**Comment by lanslotttTT** [comment]
I mean it's a normal method，like sliding window inference.



---- 回复的原邮件 ----
| 发件人 | ***@***.***> |
| 日期 | 2022年05月14日 02:39 |
| 收件人 | ***@***.***> |
| 抄送至 | ***@***.******@***.***> |
| 主题 | Re: [Project-MONAI/MONAI] How to implement sliding window training？ (Discussion #4268) |

@lanslotttTT can you share more details about sliding-window training? is it similar to the sliding-window scheme in swin transformer?
